pkg/cmd/experimental/syncgroups/groupdetector/groupdetector.go

```diff
@@ -26,6 +26,10 @@ func (l *GroupBasedDetector) Exists(ldapGroupUID string) (bool, error) {
 		return false, err
 	}
 
+	if group == nil {
+		return false, nil
```

Comment:
@liggitt amazingly it is possible that when doing a specific search for a specific DN in LDAP we can get nothing back and *not* `NoSuchObject`